src/System/Model/Types/ProductVersion.cs

```diff
@@ -109,6 +109,27 @@ public ProductVersion(int major, int minor, int build, int revision)
 			Revision = revision;
 		}
 
+		/// <summary>
+		/// Initializes a new instance of the ProductVersion class using the specified
+		/// major, minor, build and revision values.
+		/// </summary>
+		/// <param name="major">The major version number.</param>
+		/// <param name="minor">The minor version number.</param>
+		/// <param name="build">The build number (nullable)</param>
+		/// <param name="revision">The revision number (nullable)</param>
+		public ProductVersion(int major, int minor, int? build, int? revision)
```

Comment:
The other constructor with non-nullable parameters should probably be removed

Reply:
But that might a breaking change for consumers. So i kept it

Reply:
Removed it